RFS/src/main/java/com/rfs/cms/CmsClient.java

```diff
@@ -41,5 +42,46 @@ public interface CmsClient {
      * Updates the Metadata Migration entry in the CMS.  Returns an Optional; if the document was updated,
      * it will be the updated entry and empty otherwise.
      */
-    public Optional<CmsEntry.Metadata> updateMetadataEntry(CmsEntry.MetadataStatus status, String leaseExpiry, Integer numAttempts);
+    public Optional<CmsEntry.Metadata> updateMetadataEntry(CmsEntry.Metadata newEntry, CmsEntry.Metadata lastEntry);
+
+    /*
+     * Creates a new entry in the CMS for the Index Migration's progress.  Returns an Optional; if the document was
+     * created, it will be the created entry and empty otherwise.
+     */
+    public Optional<CmsEntry.Index> createIndexEntry();
+
+    /*
+     * Attempt to retrieve the Index Migration entry from the CMS, if it exists.  Returns an Optional; if the document
+     * exists, it will be the retrieved entry and empty otherwise.
+     */
+    public Optional<CmsEntry.Index> getIndexEntry();
+
+    /*
+     * Updates the Index Migration entry in the CMS.  Returns an Optional; if the document was updated,
+     * it will be the updated entry and empty otherwise.
+     */
+    public Optional<CmsEntry.Index> updateIndexEntry(CmsEntry.Index newEntry, CmsEntry.Index lastEntry);
+
+    /*
+     * Creates a new entry in the CMS for an Index Work Item.  Returns an Optional; if the document was
+     * created, it will be the created entry and empty otherwise.
+     */
+    public Optional<CmsEntry.IndexWorkItem> createIndexWorkItem(String name, int numShards);
+
+    /*
+     * Updates the Index Work Item in the CMS.  Returns an Optional; if the document was updated,
+     * it will be the updated entry and empty otherwise.
+     */
+    public Optional<CmsEntry.IndexWorkItem> updateIndexWorkItem(CmsEntry.IndexWorkItem newEntry, CmsEntry.IndexWorkItem lastEntry);
+
+    /*
+     * Forcefully updates the Index Work Item in the CMS.  This method should be used when you don't care about collisions
+     * and just want to overwrite the existing entry no matter what.  Returns the updated entry.
+     */
+    public CmsEntry.IndexWorkItem updateIndexWorkItemForceful(CmsEntry.IndexWorkItem newEntry);
+
+    /* 
+     * Retrieves a set of Index Work Items from the CMS that appear ready to be worked on, up to the specified limit.
+     */
+    public List<CmsEntry.IndexWorkItem> getAvailableIndexWorkItems(int maxItems);
```

Comment:
Why is the client changing the number of items it gets back, isn't this good enough to be in a constant?

Reply:
Hmm, I don't think it is.  It's requesting up to a maximum amount from the Target Cluster set by `maxItems` using the standard search API (for an OpenSearch implementation, at least).